### PR TITLE
cmake_install always invokes --build --target install

### PR DESCRIPTION
**Pull request by KazNX**
Removed special case code for Visual Studio generator using MSBUILD. "cmake --build" with "--target install" has the same effect and removed the need for special case code.

### CODE REVIEW

File: colcon_cmake/task/cmake/build.py [hunk 1]
```diff
@@ -313,27 +311,15 @@ def _get_make_arguments(self):
...
         self.progress('install')
 
-        generator = get_generator(args.build_base)
-        if 'Visual Studio' not in generator:
-            if CMAKE_EXECUTABLE is None:
-                raise RuntimeError("Could not find 'cmake' executable")
-            cmd = [
-                CMAKE_EXECUTABLE, '--build', args.build_base,
-                '--target', 'install']
-            job_args = self._get_make_arguments()
```
Comment (dirk-thomas) [inline colcon_cmake/task/cmake/build.py]: This line has been removed which needs to be restored.

File: colcon_cmake/task/cmake/build.py [hunk 2]
```diff
@@ -313,27 +311,15 @@ def _get_make_arguments(self):
...
-            if job_args:
-                cmd += ['--'] + job_args
```
Comment (dirk-thomas) [inline colcon_cmake/task/cmake/build.py]: These three lines have been removed which need to be restored.

Comment (KazNX) [reply colcon_cmake/task/cmake/build.py]: `job_args` restored when not using a multi-configuration generator. Seems the usage of job args is similarly dependent on the multi-configuration generator check in the build step. The installation code as is may not have worked for Xcode.

File: colcon_cmake/task/cmake/build.py [hunk 3]
```diff
@@ -313,27 +311,24 @@ def _get_make_arguments(self):
...
+        multi_configuration_generator = is_multi_configuration_generator(
+            args.build_base, args.cmake_args)
+        if multi_configuration_generator:
+            cmd += ['--config', self._get_configuration(args)]
+        else:
+            # Add make/ninja arguments for the number of job threads to use.
             job_args = self._get_make_arguments()
             if job_args:
+                # Add arguments for the number of job threads to use
+                # Arguably, _get_make_arguments() should be be configured to
```
Comment (dirk-thomas) [inline colcon_cmake/task/cmake/build.py]: If you think this should be done please consider to contribute a pull request for it. This comment in the code is unlikely to be revisited hence I will remove it.

File: colcon_cmake/task/cmake/build.py [hunk 4]
```diff
@@ -313,27 +311,24 @@ def _get_make_arguments(self):
...
+                # return arguments based on the current generator, not just for
+                # make (and ninja by coincidence). The '--' extension may be
+                # better returned from _get_make_arguments()
```
Comment (dirk-thomas) [inline colcon_cmake/task/cmake/build.py]: The `--` has nothing to do with the make arguments but with the fact that the invoked cmd is `cmake --build` and arguments which should be passed along to the specific tool need to be separated by a double dash. Therefore I will remove this comment.

**Comment by dirk-thomas** [comment]
CI builds using this branch:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/8515/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/4398/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/6932/) (known failing test)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/8428/)

**Pull request merged**